apps/lockfile-explorer/src/commands/lint.ts

```diff
@@ -0,0 +1,167 @@
+        requiredVersions,
+        checkedDependencyPaths
+      );
+    }
+  }
+}
+
+async function searchAndValidateDependencies(
+  rushConfiguration: RushConfiguration,
+  checkedProjects: Set<RushConfigurationProject>,
+  project: RushConfigurationProject,
+  requiredVersions: Record<string, string>
+): Promise<void> {
+  console.log(`Checking the project: ${project.packageName}.`);
+
+  const projectFolder: string | undefined = project.projectFolder;
+  const subspace: Subspace | undefined = project.subspace;
+  if (subspace && projectFolder) {
+    const shrinkwrapFilename: string = subspace.getCommittedShrinkwrapFilename();
+    const pnpmLockfileText: string = await FileSystem.readFileAsync(shrinkwrapFilename);
+    const doc = yaml.load(pnpmLockfileText) as Lockfile | LockfileV6;
+    const { importers, lockfileVersion, packages } = doc;
+    const shrinkwrapFileMajorVersion: number = getShrinkwrapFileMajorVersion(lockfileVersion);
+    const checkedDependencyPaths: Set<string> = new Set<string>();
+    for (const [relativePath, { dependencies }] of Object.entries(importers)) {
+      if (path.resolve(projectFolder, relativePath) === projectFolder) {
+        const dependenciesEntries = Object.entries(dependencies ?? {});
+        for (const [dependencyName, dependencyValue] of dependenciesEntries) {
+          const fullDependencyPath = `/${dependencyName}${shrinkwrapFileMajorVersion === 6 ? '@' : '/'}${
+            typeof dependencyValue === 'string'
+              ? dependencyValue
+              : (
+                  dependencyValue as {
+                    version: string;
+                    specifier: string;
+                  }
+                ).version
+          }`;
+          if (fullDependencyPath.includes('link:')) {
+            const dependencyProject: RushConfigurationProject | undefined =
+              rushConfiguration.getProjectByName(dependencyName);
+            if (dependencyProject && !checkedProjects.has(dependencyProject)) {
+              checkedProjects.add(project);
+              await searchAndValidateDependencies(
+                rushConfiguration,
+                checkedProjects,
+                dependencyProject,
+                requiredVersions
+              );
+            }
+          } else {
+            await checkVersionCompatibility(
+              shrinkwrapFileMajorVersion,
+              packages,
+              fullDependencyPath,
+              requiredVersions,
+              checkedDependencyPaths
+            );
+          }
+        }
+      }
+    }
+  }
+}
+
+async function performVersionRestrictionCheck(
+  rushConfiguration: RushConfiguration,
+  requiredVersions: Record<string, string>,
+  projectName: string
+): Promise<void> {
+  const project: RushConfigurationProject | undefined = rushConfiguration?.getProjectByName(projectName);
+  if (!project) {
+    throw new Error(`Cannot found project name: ${projectName}`);
```

Comment:
Instead of throwing here, consider having this function (and the rules) return a result object that contains failure data if a failure occurs to allow all failures to be printed instead of just the first one.

Reply:
done